src/test/java/fi/otavanopisto/kuntaapi/test/RestfulPtvServiceMocker.java

```diff
@@ -0,0 +1,264 @@
+  private static final String PHONE_CHANNELS_PATH = String.format("%s/%%s/phoneChannels", SERVICES_PATH);
+  private static final String PRINTABLE_FORM_CHANNELS_PATH = String.format("%s/%%s/printableFormChannels", SERVICES_PATH);
+  private static final String LOCATION_CHANNELS_PATH = String.format("%s/%%s/serviceLocationChannels", SERVICES_PATH);
+  private static final String WEBPAGE_CHANNELS_PATH = String.format("%s/%%s/webPageChannels", SERVICES_PATH);
+  
+  private ResourceMocker<String, Service> serviceMocker = new ResourceMocker<>();
+
+  public RestfulPtvServiceMocker() {
+    serviceMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(SERVICES_PATH));
+  }
+  
+  @Override
+  public void startMock() {
+    super.startMock();
+    serviceMocker.start();
+  }
+  
+  @Override
+  public void endMock() {
+    serviceMocker.stop();
+    super.endMock();
+  }
+  
+  /**
+   * Mocks resources
+   * 
+   * @param ids resource ids
+   * @return mocker mocker
+   */
+  public RestfulPtvServiceMocker mockServices(String... ids) {
+    try {
+      for (String id : ids) {
+        if (!serviceMocker.isMocked(id)) {
+          mockService(readServiceFromJSONFile(String.format("services/%s.json", id)));
+          
+          ResourceMocker<String, ElectronicChannel> electronicChannelMocker = new ResourceMocker<>();
+          electronicChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(ELECTRONIC_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, electronicChannelMocker);
+          
+          ResourceMocker<String, PhoneChannel> phoneChannelMocker = new ResourceMocker<>();
+          phoneChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(PHONE_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, phoneChannelMocker);
+
+          ResourceMocker<String, PrintableFormChannel> printableFormChannelMocker = new ResourceMocker<>();
+          printableFormChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(PRINTABLE_FORM_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, printableFormChannelMocker);
+
+          ResourceMocker<String, PrintableFormChannel> locationChannelMocker = new ResourceMocker<>();
+          locationChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(LOCATION_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, locationChannelMocker);
+
+          ResourceMocker<String, WebPageChannel> webPageChannelMocker = new ResourceMocker<>();
+          webPageChannelMocker.addStatusList(MockedResourceStatus.OK, urlPathEqualTo(String.format(WEBPAGE_CHANNELS_PATH, id)));
+          serviceMocker.addSubMocker(id, webPageChannelMocker);
+        } else {
+          serviceMocker.setStatus(id, MockedResourceStatus.OK);
+        }
+      }
+    } catch (JsonProcessingException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)